File: foundations/py/4/cashregister.py
def show(label, price):
    print(label,  '=', f"{round(price, 2):.2f}")

price1 = float(input('Enter price of the first item:'))
price2 = float(input('Enter price of the second item:'))
club_card = input('Does customer have a club card? (Y/N):').lower() == 'y'
tax_rate = float(input('Enter tax rate, e.g. 5.5 for 5.5% tax:'))
base_price = price1 + price2
if price1 < price2:
    discount_price = price1/2 + price2
else:
    discount_price = price1 + price2/2
if club_card:
    discount_price *= 0.9
total_price = discount_price * (1 + tax_rate/100)
show('Base price',  base_price)
show('Price after discounts', discount_price)
show('Total price', total_price)

File: foundations/cpp/10/cashregister.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main()
{
	double price1, price2, taxRate;
	char clubCard;
	cout << "Enter price of the first item: ";
	cin >> price1;
	cout << "Enter price of the second item: ";
	cin >> price2;
	cout << "Does customer have a club card? (Y/N): ";
	cin >> clubCard;
	cout << "Enter tax rate, e.g. 5.5 for 5.5% tax: ";
	cin >> taxRate;

	bool club = clubCard == 'Y' || clubCard == 'y';
	double basePrice = price1 + price2;
	double discount = price1 < price2 ? price1 / 2 : price2 / 2;
	double discountPrice = (basePrice - discount) * (club ? 0.9 : 1.0);
	double totalPrice = discountPrice * (1 + taxRate / 100);

	cout << std::fixed << std::setprecision(2)
		 << "Base price = " << basePrice << endl;
	cout << std::fixed << std::setprecision(2)
		 << "Price after discounts = " << discountPrice << endl;
	cout << std::fixed << std::setprecision(2)
		 << "Total price = " << totalPrice << endl;
	return 0;
}